python/cuml/decomposition/pca.pyx

```diff
@@ -341,15 +341,7 @@ class PCA(Base):
 
     def _build_params(self, n_rows, n_cols):
         cpdef paramsPCA *params = new paramsPCA()
-        if self.n_components is None:
-            logger.warn(
-                'Warning(`_build_params`): As of v0.16, PCA invoked without an'
-                ' n_components argument defauts to using'
-                ' min(n_samples, n_features) rather than 1'
-            )
-            params.n_components = min(n_rows, n_cols)
-        else:
-            params.n_components = self.n_components
+        params.n_components = self.n_components_
```

Comment:
I might be missing something, but where is this `n_components_` being defined? Can't find it in the file, and why is it needed in addition to `n_components`? 

Also if needed and if it is an internal variable, then `_n_components` would be a better name,

Reply:
It is defined line 433/435 on the first lines of the `fit` function, so always before the `_build_params` function is called.

This is needed because on a call to `PCA().fit(X).transform(X)`, the number of components is choosed by default during the `fit` call (because not specified by the user) but it is not saved. So the `transform` was crashing.

And using an internal variable is useful to not overwrite the user' `n_components`, just like in IPCA.

Reply:
And I changed the naming to `_n_components`